### PR TITLE
Javascript part 2 workshop

### PR DESCRIPTION
**Pull request by poe21**
Deleted my previous pull requests to have everything on one pull request / branch

### CODE REVIEW

**Comment by poe21** [comment]
Sorry for the mess..

File: addressbook.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+Finally test your code by creating a person and giving them a home and work email. 
+*/
+
+function Person(firstName, lastName) {
+    this.firstName = firstName;
+    this.lastName = lastName;
+    this.emails = []
+}
+
+function checkType(type) {
```
Comment (ziad-saab) [inline addressbook.js]: @poe21 good job on having an external function for this

File: functions.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
+// Read about the Array.prototype.sort method. Create a function that can be used with sort. 
+// This function should take two strings, and return a value that sort can use to determine which is the longest 
+// string. Finally, create an array of strings and try to get it sorted using your new function.
+
+function sentenceSorter(string1, string2) {
```
Comment (ziad-saab) [inline functions.js]: @poe21 the sorting function is actually your `compare` function. It only has to say how two elements relate to each other. The `sentenceSorter` should be able to take an array of as many strings as you like, not only two.

File: warriors.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        return "F";
+    }
+}
+
+function Warrior(name) {
+    this.name = name;
+    this.level = 1;
+    this.weapon = "wooden sword";
+    this.power = (Math.random() * (101 - 1)) + 1;
+    this.gender = getRandomGender(); // for fun :)))
```
Comment (ziad-saab) [inline warriors.js]: :sunglasses:

**Pull request closed without merging**